### PR TITLE
Return proper error if part file name is too long

### PR DESCRIPTION
**Pull request by JammingBen**
## Description
This is an alternative, less destructive approach than https://github.com/owncloud/core/pull/39088.

With this implementation, we keep the filename limit for part file names and just return a proper error. That means, if you upload a file with 250 chars (filesystem limit is usually 255), then you still get an "filename too long" error because of the part file name added by ownCloud.

This PR also reverts https://github.com/owncloud/core/pull/39088 therefore.

## Related Issue
- Fixes https://github.com/owncloud/enterprise/issues/4692

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [x] Code changes
- [x] Unit tests added
- [ ] Acceptance tests added
- [x] Documentation ticket raised: https://github.com/owncloud/docs/issues/3994
- [x] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

**Comment by JammingBen** [comment]
@mrow4a @DeepDiver1975 As discussed, here is the alternative approach for our part file name issue. Please have a look.

**Comment by JammingBen** [comment]
Please note that by setting `part_file_in_storage` to `false`, part files will still get hashed (see https://github.com/owncloud/core/blob/master/apps/dav/lib/Connector/Sabre/File.php#L310)! That means this setting is basically broken because, once again, several apps cannot work with a hashed part file name yet. https://github.com/owncloud/core/pull/39131 would solve this problem though.

**Comment by DeepDiver1975** [comment]
I see no change from 255 to 250 in this pr.
Am I missing anything? Thx

**Comment by JammingBen** [comment]
Changing this won't affect anything, because the part file is being written to the filesystem before the length check of the original file. Thus we just added a length check for the part file itself.

File: changelog/unreleased/39088 [hunk 1]
```diff
@@ -1,10 +0,0 @@
-Bugfix: Hash chunk(v2) filename
```
Comment (mrow4a) [inline changelog/unreleased/39088:1]: I think we still need changelog, don't we?

**Review by mrow4a — COMMENTED** [review]
add quick unit test

**Review by DeepDiver1975 — APPROVED** [review]

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiProvisioning-v1-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/32162/53/1

**Review by mrow4a — APPROVED** [review]

File: config/config.sample.php [hunk 2]
```diff
@@ -1498,6 +1498,9 @@
  * same storage as the upload target. Setting this to false will store the part
  * files in the root of the users folder which might be required to work with certain
  * external storage setups that have limited rename capabilities.
+ *
+ * Note that this setting causes issues with the following apps: Encryption,
```
Comment (mmattel) [inline config/config.sample.php:1502]: It is not clear if the issue is when the setting is default (true) or when changing it to false.

**Comment by phil-davis** [comment]
https://drone.owncloud.com/owncloud/core/32163
```
There was 1 failure:

1) OCA\DAV\Tests\unit\Connector\Sabre\FileTest::testPutTooLongPartFileName
Failed asserting that exception of type "OCA\DAV\Connector\Sabre\Exception\FileNameTooLong" is thrown.
```

https://drone.owncloud.com/owncloud/core/32163/32/10
```
There was 1 failure:

1) OCA\DAV\Tests\unit\Connector\Sabre\FileTest::testPutTooLongPartFileName
Failed asserting that exception of type "OCA\DAV\Connector\Sabre\Exception\FileNameTooLong" is thrown.
```

These are both errors when running with `scality` storage. My guess is that `scality` storage allows longer file names, and so the exception is not thrown in that case.

I am not sure how you can get the file system max filename length from within PHP. If you could do that, then the test could generate a filename a bit longer than the maximum allowed by the current storage. Or maybe tag this test case to skip on scality (however that is done in the unit tests).

**Comment by mmattel** [comment]
> I am not sure how you can get the file system max filename length from within PHP

Maybe this can help a bit...
https://doc.owncloud.com/server/admin_manual/troubleshooting/path_filename_length.html

**Comment by JammingBen** [comment]
This is not the issue I think as the character limit is hard coded: https://github.com/owncloud/core/blob/master/lib/private/Files/Storage/Common.php#L507. It rather seems that the scality storage does not use part files at all, therefore no exception. Let's see if it succeeds now.

**Comment by phil-davis** [comment]
All the failing CI in oC10 apps is passing now. This core change was "a good thing".

**Comment by jnweiger** [comment]
### Tested with owncloud core 10.9.0 beta1

228 characters is the limit on my https://oc1090beta1-ldap-0160-rc2-20211126.jw-qa.owncloud.works testsystem. A file with 229 chars triggers a "filename too long" error. OKayish.

**Pull request merged**